src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1107,6 +1097,18 @@ protected override void PreStart()
             _ackIdleTimerCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(ackIdleInterval, ackIdleInterval, Self, AckIdleCheckTimer.Instance, Self);
         }
 
+        private async Task<object> AssociateAsync()
+        {
+            try
+            {
+                return new Handle(await Transport.Associate(RemoteAddress, _refuseUid));
+            }
+            catch (Exception e)
+            {
+                return new Status.Failure(e.InnerException);
```

Comment:
We still get the AggregatedException here because of PipeTo - hence we take the inner exception, which is either InvalidAssociationException or some unhandled exception - in any case we propagate this further.